### PR TITLE
class-extension

### PR DESCRIPTION
**Pull request by heatonk**
Adding functionality to Pathfinder to support more underlying structures, including software, services, and operating systems.

### CODE REVIEW

**Comment by elegantmoose** [comment]
@mrengstrom - These are new fields and types for a new scan report source. Doesnt not take any existing object/fields away just adds to them.

File: .gitignore [hunk 1]
```diff
@@ -11,3 +11,4 @@
 !.logs/.gitkeep
 data/reports/**
 data/adversaries/**
+app/parsers/**
```
Comment (mrengstrom) [inline .gitignore:14]: The parsers are considered source code, so they should be tracked in source control, any reason to ignore them?

**Comment by mrengstrom** [comment]
I left one comment about the `.gitignore` change, otherwise everything else looks good 👍

File: .gitignore (hunk 1 above)
Comment (elegantmoose) [reply .gitignore:14]: Im guessing @heatonk did that to make sure we didnt upload a parser that is still not public, and then forgot to take it out.

Ill remove it real quick.

Comment (elegantmoose) [reply .gitignore]: @mrengstrom - should be good now. Good catch.

Comment (mrengstrom) [reply .gitignore]: Thanks, and sounds good 👍

**Pull request merged**